tests/h/presenters/annotation_json_test.py

```diff
@@ -1,43 +1,16 @@
 import datetime
 from unittest import mock
+from unittest.mock import create_autospec
 
 import pytest
 from pyramid import security
 from pyramid.authorization import ACLAuthorizationPolicy
-from zope.interface import implementer
 
-from h.formatters.interfaces import IAnnotationFormatter
+from h.formatters import AnnotationFormatter
 from h.presenters.annotation_json import AnnotationJSONPresenter
 from h.traversal import AnnotationContext
 
 
-@implementer(IAnnotationFormatter)
-class FakeFormatter:
-    def __init__(self, data=None):
-        self.data = data or {}
-
-    def preload(self, ids):
-        pass
-
-    def format(self, annotation):
-        return self.data
-
-
-@implementer(IAnnotationFormatter)
-class IDDuplicatingFormatter:
-    """This formatter take the annotation's ID and adds it in another key.
-
-    The main purpose of it is to confirm that the presenter is passing in the
-    AnnotationContext object.
-    """
-
-    def preload(self, ids):
-        pass
-
-    def format(self, annotation_resource):
-        return {"duplicated-id": annotation_resource.annotation.id}
```

Comment:
Nice to see these gone. I don't like to see fake classes in our tests. I prefer mocks with autospeccing. Partially because mocks have nice ways of making them return different things or raise different things etc (they avoid programmers writing random logic in fake classes). And partially because of autospeccing.

The use of `@implementer` actually made these less bad than most fake classes since it would prevent the fakes from getting out of date. Similar can be done with ABCs or with type checking.